### ISSUE TITLE
How to add a term to an expression

### ISSUE THREAD

**Issue by giadarol** [xsuite/xdeps#2]
I have tried these ways but they both give the wrong result:
```python
import xdeps as xd

dct = {'a':1, 'b':2, 'c':3}
manager = xd.Manager()
dref = manager.ref(dct, 'dct')
dref['a'] = 10
dref['b'] = 20
dref['c'] = 30
assert dref['a']._get_value() == 10
dref['sum'] = dref['a'] + dref['b']
assert dref['sum']._get_value() == 30
dref['sum'] += dref['c']
print(dref['sum']._value) # this gives 90 (should give 90)


dct = {'a':1, 'b':2, 'c':3}
manager = xd.Manager()
dref = manager.ref(dct, 'dct')
dref['a'] = 10
dref['b'] = 20
dref['c'] = 30
assert dref['a']._get_value() == 10
dref['sum'] = dref['a'] + dref['b']
assert dref['sum']._get_value() == 30
manager.tasks[dref['sum']].expr += dref['c']
print(dref['sum']._value) # this gives 30 (should give 60)
dref['a'] = 10
print(dref['sum']._value) # this gives 30 (should give 60) 
```

**Comment by rdemaria** [comment]
Quick reply without testing and without looking at the code, I might say something wrong.

The first approach cannot work because it is equivalent to write in madx sum:=sum + c

The second approach cannot work because it changes the expression of the task but it does register the new dependencies.

The approach that should work is:

```
newexpr = manager.tasks[dref['sum']].expr + dref['c']
dref['sum']=newexpr
```

I understand that this is cumbersome and a convenient API should be added but it needs some thoughts (probably first approach should be feasible but might be more confusing)


I have also noticed a potential bug when overwriting a previous definition. This needs some careful testing...

**Comment by rdemaria** [comment]
Fixed by 589045c
